### PR TITLE
Add tests for CachingLinter

### PR DESCRIPTION
**Pull request by ntzm**

### CODE REVIEW

File: tests/Linter/CachingLinterTest.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            'bar.php' => '<?php echo "baz";',
+            'baz.php' => '<?php echo "foobarbaz";',
+        ));
+
+        $result = $this->prophesize('PhpCsFixer\Linter\LintingResultInterface');
+
+        $sublinter = $this->prophesize('PhpCsFixer\Linter\LinterInterface');
+        $sublinter->lintFile($fs->url().'/foo.php')->shouldBeCalledTimes(1)->willReturn($result->reveal());
+        $sublinter->lintFile($fs->url().'/bar.php')->shouldNotBeCalled();
+        $sublinter->lintFile($fs->url().'/baz.php')->shouldBeCalled()->willReturn($result->reveal());
```
Comment (keradus) [inline tests/Linter/CachingLinterTest.php]: why we count number of calls to foo and bar, but not baz ?

File: tests/Linter/CachingLinterTest.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+        $linter = new CachingLinter($sublinter->reveal());
+
+        $linter->lintFile($fs->url().'/foo.php');
```
Comment (julienfalque) [inline tests/Linter/CachingLinterTest.php]: Please add assertions on the result of `lintFile()` calls here and `lintSource()` in the following test.

File: tests/Linter/CachingLinterTest.php (hunk 1 above)
Comment (ntzm) [reply tests/Linter/CachingLinterTest.php]: Will update

File: tests/Linter/CachingLinterTest.php (hunk 2 above)
Comment (ntzm) [reply tests/Linter/CachingLinterTest.php]: Will do, thanks 👍

File: tests/Linter/CachingLinterTest.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        $linter = new CachingLinter($sublinter->reveal());
+
+        $results = array(
+            $linter->lintFile($fs->url().'/foo.php'),
+            $linter->lintFile($fs->url().'/foo.php'),
+            $linter->lintFile($fs->url().'/bar.php'),
+            $linter->lintFile($fs->url().'/baz.php'),
+        );
+
+        $this->assertContainsOnlyInstancesOf('PhpCsFixer\Linter\LintingResultInterface', $results);
```
Comment (julienfalque) [inline tests/Linter/CachingLinterTest.php]: This assertion is nice but not sufficient: the `CachingLinter` may return any instance of `LintingResultInterface` without making the assertion fail. You should check that each call returns `$result`, which would prove the `CachingLinter` just returns the result provided by the decorated `$sublinter`.

Maybe that would require to mock several results though, as you call `lintFile()` on different files.

Comment (ntzm) [reply tests/Linter/CachingLinterTest.php]: Thanks, will update

**Comment by keradus** [comment]
Thanks @ntzm !
Merged manually into #3723

**Pull request closed without merging**